### ISSUE TITLE
Add support for telepathy-haze (empathy)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [EionRobb/skype4pidgin#51]
```
It would be great if the plugin could be used in empathy. To my knowledge,
it is possible to use purple plugins with empathy, if there are appropriate
sections in haze.manager and a mission-control profile.

See https://bugs.freedesktop.org/show_bug.cgi?id=17907
```

Original issue reported on code.google.com by `florian....@gmail.com` on 13 Mar 2009 at 9:02

**Comment by rodneyrod** [comment]
So should this workaround work with SkypeWeb?

**Comment by EionRobb** [comment]
@rodneyrod yes, the SkypeWeb plugin works with telepathy-haze.  No workaround needed as far as I know

**Comment by CapsAdmin** [comment]
I installed pidgin, built and installed SkypeWeb as well as telepathy-haze on the fedora kde spin but I'm unable to find it in empathy's online account section. It does show and work in pidgin though.

**Comment by rodneyrod** [comment]
Yeah can we open this issue again? I've done make && make install on a distro with telepathy-haze and no pidgin and I'm getting no extra connections showing in the connection manager.

**Comment by EionRobb** [comment]
@CapsAdmin @rodneyrod can you get a libpurple debug log as haze loads in the plugins?  It should say something about missing a dependency or something and should help point us to what's going wrong

**Comment by rodneyrod** [comment]
Still trying to work out how to get this working with Telepathy Haze, haven't really been able to get a debug log out and it isn't easy to do at KDE since everything is loaded at boot AFAIK.

Also, someone here started porting your work to Sailfish OS, and with a bit of work could be ported back to the desktop versions of Telepathy, but since I know 0% of qml or C, there's not much I can do to help coding wise
https://github.com/Thaodan/haze-accounts-extensions-skypeweb

**Comment by EionRobb** [comment]
I don't know anything about Telepathy nor Haze unfortunately, but I have heard of other users using it successfully.

Unfortunately without a debug log I can't really give any advice on what might be happening or what the next steps might be.  All I can suggest is ensuring all the dependencies exist (libpurple, glib2, json-glib), but that's probably not that helpful in this case, sorry :(

**Comment by dionorgua** [comment]
I've got it working on my laptop. Without any changes to code. The only thing that is missing is 'groupchats'. 
Pidgin provides a way to search for room and join it. In Empathy it's disabled for skype account.

**Comment by rodneyrod** [comment]
@dionorgua Could you provide any instructions that you have? Also was this on Telepathy or TelepathyQt?

**Comment by dionorgua** [comment]
It works at the same time with KDE telepathy (from debian experimental) and Empathy,

No instructions actually. I've just compiled it, created deb package and installed. Then added 'skype (HTTP)' account.

**Comment by rodneyrod** [comment]
So when you say created a deb package, what steps did you follow?
I just tried 'sudo make install' and got no luch there.

**Comment by dionorgua** [comment]
I've just followed steps from README. basically cmake, make and then cpack.
I think make install installs files to same place.

**Comment by dionorgua** [comment]
Just make sure you've /usr/lib/purple-2/libskypeweb.so

**Comment by rodneyrod** [comment]
I'm running Netrunner (Manjaro/Arch) with KDE and Telepathy and I've had no luck. The library is in the right place, but it just isn't working for whatever reason.

**Comment by dionorgua** [comment]
How it doesn't work? You can't add account? Can't authorize or what?

**Comment by dionorgua** [comment]
How it doesn't work? Can't add account, can't authorize?

Also try to use Empathy.

**Comment by rodneyrod** [comment]
Yeah I can't add the account, it isn't shown in the Online Accounts list in KDE, which is where you manage Telepathy accounts in KDE.

**Comment by dionorgua** [comment]
Just add it via Empathy. Otherwise you need something like this:
https://github.com/Thaodan/haze-accounts-extensions-skypeweb

**Comment by rodneyrod** [comment]
Empathy pulls in too many Gnome libraries to be of any use for me personally, and that shim only works on SailfishOS and no one seems to be interested in porting it to the desktop.

Isn't there a way to get this working through KDE's settings manager?

**Comment by rodneyrod** [comment]
Empathy pulls in too many Gnome libraries to be of any use for me personally, and that shim only works on SailfishOS and no one seems to be interested in porting it to the desktop.

Isn't there a way to get this working through KDE's settings manager?

**Comment by khurshid-alam** [comment]
@dionorgua I installed it webupd8 ppa (http://www.webupd8.org/2016/07/chat-with-your-skype-friends-from.html), but when I run `empathy-accounts`, I don't see the account. I am using Ubuntu 16.04 with Ubuntu-Online-Accounts which doesn't expose telepathy haze accounts. So my only option is to add it via `empathy-accounts`. I can see all other purple account except skype.

**Comment by dionorgua** [comment]
Sorry. I don't know why it doesn't work and doesn't show for you. There were no additional steps from my site except building libskypeweb and installing it to /usr/lib/purple-2/libskypeweb.so

Don't forget to restart mission-control after this (or just reboot to be sure).

![skypeweb](https://cloud.githubusercontent.com/assets/534357/17112384/6037c62c-5295-11e6-9d8d-e28991993df9.png)

**Comment by khurshid-alam** [comment]
@dionorgua 

Ah. Thanks. I made it working by running `empathy-accounts`. It connects but not all contacts or bots are visible. But I guess that's a temporary issue.

![telepathy-haze-skypeweb-empathy](http://i.imgur.com/7P0Rcfq.png)
